src/toil_scripts/rnaseq_cgl/rnaseq_cgl_pipeline.py

```diff
@@ -201,12 +202,21 @@ def process_sample(job, config, input_tar=None, input_r1=None, input_r2=None, gz
     for root, subdir, files in os.walk(work_dir):
         fastqs.extend([os.path.join(root, x) for x in files])
     if config.paired:
-        r1 = sorted([x for x in fastqs if 'R1' in x])
-        r2 = sorted([x for x in fastqs if 'R2' in x])
-        if not r1 or not r2:
-            r1 = sorted([x for x in fastqs if '_1' in x])
-            r2 = sorted([x for x in fastqs if '_2' in x])
-        require(len(r1) == len(r2), 'Check fastq naming, uneven number of pairs found: r1: {}, r2: {}'.format(r1, r2))
+        r1, r2 = [], []
+        # Pattern convention: Look for R1/R2 or _1/_2 at the end of files named .fastq.gz, .fastq, .fq.gz, or .fq
+        pattern = re.compile('(?:^|[._-])R?([12])(\\.fastq(\\.gz)?|\\.fq(\\.gz)?)$')
+        for fastq in sorted(fastqs):
+            match = pattern.search(os.path.basename(fastq))
+            if not match:
+                raise UserError('FASTQ file name fails to meet required convention for paired reads '
+                                '(see documentation). ' + fastq)
+            elif match.group(1) == '1':
+                r1.append(fastq)
+            elif match.group(1) == '2':
+                r2.append(fastq)
+            else:
+                assert False
```

Comment:
At the least, the error message should indicate what the filename that failed in the regex was.